openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -655,14 +657,17 @@ def update_select_courses(cls, course_keys, force_update=False):
         log.info('Finished generating course overviews.')
 
     @classmethod
-    def get_all_courses(cls, orgs=None, filter_=None):
+    def get_all_courses(cls, orgs=None, filter_=None, **kwargs):
         """
         Return a queryset containing all CourseOverview objects in the database.
 
         Arguments:
             orgs (list[string]): Optional parameter that allows case-insensitive
                 filtering by organization.
             filter_ (dict): Optional parameter that allows custom filtering.
+            kwargs (dict): Optional key value pairs usable for performing complex lookups on course listing.
+                * Possible values:
+                    * active_courses_only (bool): If provided, only the courses that have not ended will be returned.
```

Comment:
Similarly here, I think changing this to an explicit kwarg is going to be a clearer, self-documenting design.

```suggestion
    def get_all_courses(cls, orgs=None, filter_=None, active_only=False):
        """
        Return a queryset containing all CourseOverview objects in the database.

        Arguments:
            orgs (list[string]): Optional parameter that allows case-insensitive
                filtering by organization.
            filter_ (dict): Optional parameter that allows custom filtering.
            active_only (bool): If True, only the courses that have not ended will be returned.
```